sympy/sets/fancysets.py

```diff
@@ -334,66 +385,83 @@ def __new__(cls, *args):
 
         # expand range
         slc = slice(*args)
+        if slc.step == 0:
+            raise ValueError("step cannot be 0")
         start, stop, step = slc.start or 0, slc.stop, slc.step or 1
         try:
-            start, stop, step = [w if w in [S.NegativeInfinity, S.Infinity] else sympify(as_int(w))
-                                 for w in (start, stop, step)]
+            start, stop, step = [
+                w if w in [S.NegativeInfinity, S.Infinity]
+                else sympify(as_int(w))
+                for w in (start, stop, step)]
         except ValueError:
             raise ValueError("Inputs to Range must be Integer Valued\n" +
                     "Use ImageSets of Ranges for other cases")
 
         if not step.is_finite:
             raise ValueError("Infinite step is not allowed")
-        if start == stop:
-            return S.EmptySet
 
-        n = ceiling((stop - start)/step)
-        if n <= 0:
-            return S.EmptySet
+        if all(i.is_infinite for i in  (start, stop)):
+            raise ValueError("The start and end value of "
+                             "Range cannot both be unbounded")
```

Comment:
Since we have deprecated `bounded/unbounded` in favour of `finite/infinite` in assumptions, should we change it to `infinite` here? So that we can maintain same things in all places? 
Just a suggestion. Let me know what you think.